renku/core/commands/session.py

```diff
@@ -0,0 +1,177 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku session commands."""
+
+import webbrowser
+from itertools import chain
+from typing import Optional
+
+from yaspin import yaspin
+
+from renku.core import errors
+from renku.core.commands.format.session import SESSION_FORMATS
+from renku.core.management.command_builder import inject
+from renku.core.management.command_builder.command import Command
+from renku.core.management.interface.client_dispatcher import IClientDispatcher
+from renku.core.models.session import ISessionProvider
+from renku.core.plugins.session import supported_session_providers
+from renku.core.utils import communication
+from renku.core.utils.os import safe_read_yaml
+
+
+@inject.autoparams()
+def _get_renku_project_name(client_dispatcher: IClientDispatcher) -> str:
+    client = client_dispatcher.current_client
+    return f"{client.remote['owner']}/{client.remote['name']}" if client.remote["name"] else f"{client.path.name}"
+
+
+def _safe_get_provider(provider: str) -> ISessionProvider:
+    providers = supported_session_providers()
+    p = next(filter(lambda x: x[1] == provider, providers), None)
+
+    if p is None:
+        raise errors.ParameterError(f"Session provider '{provider}' is not available!")
+    return p[0]
+
+
+def _session_list(config_path: str, format="tabular", provider: Optional[str] = None):
+    project_name = _get_renku_project_name()
+    config = safe_read_yaml(config_path) if config_path else dict()
+
+    providers = supported_session_providers()
+    if provider:
+        providers = list(filter(lambda x: x[1] == provider, providers))
+
+    if len(providers) == 0:
+        raise errors.ParameterError("No session provider is available!")
+
+    sessions_list = list(chain(*map(lambda x: x[0].session_list(config=config, project_name=project_name), providers)))
+    return SESSION_FORMATS[format](sessions_list)
+
+
+def session_list_command():
+    """List all the running interactive sessions."""
+    return Command().command(_session_list)
+
+
+@inject.autoparams("client_dispatcher")
+def _session_start(
+    provider: str,
+    config_path: str,
+    client_dispatcher: IClientDispatcher,
+    image_name: str = None,
+    cpu_request: Optional[float] = None,
+    mem_request: Optional[str] = None,
+    disk_request: Optional[str] = None,
+    gpu_request: Optional[str] = None,
+):
+    client = client_dispatcher.current_client
+
+    project_config = client.load_config()
+    pinned_image = project_config.get("interactive", "image", fallback=None)
```

Comment:
```suggestion
    pinned_image = client.get_value("interactive", "image")
```
Also for other places where config is used.